### PR TITLE
CSP-112 Implements national delivery question

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditNationalDeliveryOptionControllerTests/EditNationalDeliveryOptionGetValidSessionTests.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            ViewResult.Should().NotBeNull();
+        }
+
+        [Test]
+        public void ThenSetViewModelOnTheResult()
+        {
+            Model.Should().NotBeNull();
+        }
+
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditNationalDeliveryOptionControllerTests/EditNationalDeliveryOptionGetValidSessionTests.cs:35]: suggestion - add a test for larscode, just so there is another vanilla model mapping test
Model.LarsCode.Should().Be(LarsCode); 
I think is correct.

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            _sessionService = sessionService;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        [Route("{ukprn}/standards/{larsCode}/edit-national-delivery-option", Name = RouteNames.GetNationalDeliveryOption)]
+        public IActionResult Index([FromRoute] int larsCode)
+        {
+            if(!IsCorrectLocationOptionSetInSession(larsCode))
+            {
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs:35]: add log

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                return RedirectToRoute(RouteNames.GetLocationOption, new { Ukprn, larsCode });
+            }
+
+            var model = new EditNationalDeliveryOptionViewModel();
+            model.BackLink = Url.RouteUrl(RouteNames.GetLocationOption, new { Ukprn, larsCode });
+            model.CancelLink = GetStandardDetailsUrl(larsCode);
+            return View(model);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs:42]: add log

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                return View(model);
+            }
+
+            if (model.HasNationalDeliveryOption.GetValueOrDefault())
+            {
+                await _mediator.Send(new AddNationalLocationToStandardCommand(Ukprn, model.LarsCode, UserId));
+                return RedirectToRoute(RouteNames.GetStandardDetails, new { Ukprn, model.LarsCode });
+            }
+
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs:59]: add log

**Review by MarkFCain — DISMISSED** [review]
one suggestion for adding a test, and EditNationalDeliveryOptionController needs logs.  Approved.

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditNationalDeliveryOptionControllerTests/EditNationalDeliveryOptionGetValidSessionTests.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditNationalDeliveryOptionControllerTests/EditNationalDeliveryOptionGetValidSessionTests.cs:35]: LarsCode is not populated on the model during get, as I dont need it and I am reading it from the route during post.

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("{ukprn}/standards/{LarsCode}/edit-national-delivery-option", Name = RouteNames.PostNationalDeliveryOption)]
+        public async Task<IActionResult> Index(EditNationalDeliveryOptionViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs:51]: Added

File: src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs (hunk 4 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/EditNationalDeliveryOptionController.cs:59]: Added

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — DISMISSED** [review]

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**